ion/textutils.go

```diff
@@ -319,3 +319,7 @@ func parseInt(str string, radix int) (interface{}, error) {
 func parseTimestamp(val string) (Timestamp, error) {
 	return ParseTimestampFromStr(val)
 }
+
+func isStringValueNil(errMsg string) bool {
+	return errMsg == "ion: usage error in Reader.StringValue: string value is unknown"
```

Comment:
I think this is a strong sign that we have an API that should be returning `*string` instead of `string`. Requiring users to do a string comparison against an error message that could change without notice is very brittle.

Reply:
Updated stringvalue() to return `(*string, error)`